test/core/event_engine/test_suite/client_test.cc

```diff
@@ -114,46 +114,40 @@ TEST_F(EventEngineClientTest, ConnectExchangeBidiDataTransferTest) {
   auto quota = grpc_core::ResourceQuota::Default();
   args = args.Set(GRPC_ARG_RESOURCE_QUOTA, quota);
   ChannelArgsEndpointConfig config(args);
-  auto status = oracle_ee->CreateListener(
+  auto listener = oracle_ee->CreateListener(
       std::move(accept_cb),
       [](absl::Status status) { GPR_ASSERT(status.ok()); }, config,
       std::make_unique<grpc_core::MemoryQuota>("foo"));
-  EXPECT_TRUE(status.ok());
+  ASSERT_TRUE(listener.ok());
 
-  std::unique_ptr<Listener> listener = std::move(*status);
-  EXPECT_TRUE(listener->Bind(URIToResolvedAddress(target_addr)).ok());
-  EXPECT_TRUE(listener->Start().ok());
+  ASSERT_TRUE((*listener)->Bind(URIToResolvedAddress(target_addr)).ok());
```

Comment:
This can be `ASSERT_TRUE(listener->Bind(...).ok())`, no need for `(*listener)->`. Part of the reason it's recommend to name StatusOr variables as the value types themselves is because the StatusOr class overrides `operator->`, allowing the variable to act as the type.

This pattern happens again below.

Reply:
Somehow it doesn't compile when I do that. I de-referenced the StatusOr now immediately when its created.